protoc-gen-openapiv2/internal/genopenapi/template_fuzz_test.go

```diff
@@ -0,0 +1,19 @@
+//go:build go1.18
+// +build go1.18
+
+package genopenapi
+
+import (
+	"testing"
+)
+
+func FuzzPopulateQueryParameters(f *testing.F) {
+	f.Add("Text\n\n(-- Comment --)\n\nMore Text\n")
+	f.Add("Text\n\n(-- Multi\nLine\n\nComment --)\n\nMore Text\n")
+	f.Add("(-- Starting with comment --)\n\nMore Text\n")
+	f.Add("\n\n(-- Starting with new line and comment --)\n\nMore Text\n")
+	f.Add("Ending with\n\n(-- Comment --)")
+	f.Fuzz(func(t *testing.T, s string) {
+		_ = removeInternalComments(s)
+	})
```

Comment:
Thanks for adding the fuzz test! I think this is a good starting point, but as the bot suggests, we actually should add some logic here. Funnily enough, the bot suggested a regex solution which is "obviously correct", and I think we can use that to _differentially fuzz_ your implementation. That is, define the regex globally in this test file:

```go
var replaceInternalCommentsRegex = regexp.MustCompile(`\(\--.*?--\)`)
```

Then in the fuzz test, run both and compare the output - it should always be the same, or we have a bug:

```go
	f.Fuzz(func(t *testing.T, s string) {
		s1 := removeInternalComments(s)
		s2 := replaceInternalCommentsRegex.ReplaceAllString(s, "")
		if s1 != s2 {
			t.Errorf("Unexpected comment removal difference: our function produced %q but regex produced %q", s1, s2)
	})
```

This will allow us to test our specialized implementation against one that is obviously correct, in the same way that you could test a custom sorting algorithm against a simple but inefficient algorithm like bubble sort.

Reply:
Done, regex a bit more complicated - but at least I found some small bug on the way